src/core/public/http/anonymous_paths.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { HttpSetup } from 'src/core/public';
+
+export class AnonymousPaths {
+  private paths: Set<string>;
+
+  constructor(private basePath: HttpSetup['basePath']) {
+    this.paths = new Set();
+  }
+
+  public isAnonymous(path: string): boolean {
+    return this.paths.has(this.basePath.remove(path));
+  }
+
+  public register(path: string) {
+    this.paths.add(path);
```

Comment:
Do we want any kind of validation on the registered path? For example, should paths start with a `/`, and not end with one?

Reply:
Good point, better validation forthcoming.